connexion/apis/abstract.py

```diff
@@ -262,18 +364,19 @@ def add_operation(self, method, path, swagger_operation, path_parameters):
             "pass_context_arg_name": self.pass_context_arg_name
         }
 
-        if self.spec_version < (3, 0, 0):
+        # TODO refactor into AbstractOperation.from_spec(Specification, method, path)
```

Comment:
> Can't we just pass the specification object as a whole?

This way it will create the right `Operation` from the right `Specification` without needing the explicit version check here.

What do you think?